### ISSUE TITLE
May Monthly Payments to add to system

### ISSUE THREAD

**Issue by kbt1** [omnitekinc/AERA_MSS_Clients#159]
here are the May monthly payments.
a few have quantity of 2 which means 1 payment is for previous month and 1 is for current month.

Once this is done, can all Monthly Payment Records be set to 6/1/17? This way starting June, i will be adding all monthly payments in Dues going forward. These batches could potentially be 400-500 entries.
I will create a batch of just monthly charges and then i will create a batch of monthly checking.
**It would be nice to be able to copy a Monthly Batch and just be able to edit entries for declines. Think about it if this can be done.** 

[May Monthly Payments to be added to New Membership System..xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1026382/May.Monthly.Payments.to.be.added.to.New.Membership.System.xlsx)

**Comment by sagarjagdish** [comment]
We have loaded provided excel file into system with "Auto Entry May" prefix. however need more clarification on below point. 

@omnitekinc  :  can you please explain ?
+++++++++++++++
It would be nice to be able to copy a Monthly Batch and just be able to edit entries for declines. Think about it if this can be done.
+++++++++++++++

**Comment by kbt1** [comment]
I dont know if you can do this or not or if I can create it and keep it for future batches on a monthly basis.
I have to enter the monthly payments in the system each month. For Credit Card Transactions, that could be over 400 entries. It is going to be a long process and if I can make it easier for myself, it would help.
I will have to do the checking transactions also, however, we have fewer of those but once again, it would be nice to have a prepopulated dues batch so all I have to do is add new records or delete transactions that were declined.

Here is a sample of the entry for May Payments. 
This report does not include manual entries from declined credit cards.

[Monthly Transactions.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1029442/Monthly.Transactions.xlsx)

**Comment by omnitekinc** [comment]
@kbt1 Karen, we will discuss this today on the call and will come up with instruction for Sagar.

**Comment by kbt1** [comment]
I just need to simplify the monthly payment entries otherwise, it will take quite a while to enter them all in the system.

**Comment by sagarjagdish** [comment]
@omnitekinc :

I think here we can provide bulk insert for dues transaction. we can provide option to upload excel file and that excel file will create batch and transaction under that batch as Open batch. 

AERA need to review those transaction inside open batch and if they feel everything okay then they need to "Post" that batch.

Please let me know this approach will work so I can work on excel template and give it to you/Karen to review.

**Comment by kbt1** [comment]
That sounds perfect.
I need to be able to add and edit the batch and name it also.
Can we select from charges and a seperate checking batch. I process them on seperate days.
Charges clear the bank in a couple days, checking takes 2 weeks to clear.


I have attached my most recent list. Let me know if you have any questions.

[Current Monthly Payments for Stella for Dues Batches..xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1039170/Current.Monthly.Payments.for.Stella.for.Dues.Batches.xlsx)

**Comment by omnitekinc** [comment]
@sagarjagdish Can we finish this function by tomorrow, Karen would like to generate the report this week because she is on vacation next week.

**Comment by kbt1** [comment]
Stella and Sagar, here is the most current list of Monthly Payments for the 2 Batches that will be created.
[2015 to 2016 monthly payments spreadsheet Most Current List.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1055683/2015.to.2016.monthly.payments.spreadsheet.Most.Current.List.xlsx)

**Comment by kbt1** [comment]
all exp dates 6/1/17
Build in 30 days buffer

**Comment by omnitekinc** [comment]
@sagarjagdish Please change all these members in the Excel expiration date to June 1st for both membership and Prosis Pro. 

Can we also add a function for Prosis Pro: Adding a 30 days buffer for Prosis Pro logout date. Currently, if the monthly Prosis Pro user expired on June 1st, they will not be able to login to Prosis Pro. This should not change the expiration date, it just giving the user additional time to pay for their monthly without getting logout.

**Comment by sagarjagdish** [comment]
We have started work on this and it will be completed before end of this week. I have attached EXCEL template which will accept by System to Bulk dues create process. Please review template and let me know you suggestion if any.

attached file having 2 sheet as below.
1) Instruction = having information of all those column which listed in 2nd sheet with sample data.
2) Monthly-Payment-Duestransaction = list of transaction which we need to load, system will process only on this sheet.

[Bulk_DuesEntry.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1058778/Bulk_DuesEntry.xlsx)

**Comment by kbt1** [comment]
i have made notes on the Bulk Dues Entry spreadsheet and attached it.
[Bulk_DuesEntry - corrections.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1058871/Bulk_DuesEntry.-.corrections.xlsx)

i have also attached most current spreadsheet
[2015 to 2016 monthly payments spreadsheet for Sagar 060717.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1058872/2015.to.2016.monthly.payments.spreadsheet.for.Sagar.060717.xlsx)

please have spreadsheet sort by ID#

**Comment by kbt1** [comment]
Stella and Sagar:
Here is what I need to enter in New System for Credit Card Monthly Payments.
I ran the credit cards and entered in QuickBooks and now I will need to enter in Dues Processing.
I wanted to give you an exact sample of what I need to enter so you can see the spreadsheet.
This is exactly how I need the batch spreadsheet to look like.
I had 22 declined which are not on this report.
This is just the credit cards. The checking will be done in 2 weeks.

[Credit Card Monthly Payments for New System.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1061725/Credit.Card.Monthly.Payments.for.New.System.xlsx)

**Comment by sagarjagdish** [comment]
@kbt1 - I have below question with the previously attached file please see below.

1) What it the use of Total for # and TOTAL, when we are loading that data into Dues Processing?
2) What is the use of QTY and how I need to deal with that during auto Dues Processing?

I am fine with the other data and as per your new format we need to make few change in our current working implementation as we already started work on this.

**Comment by kbt1** [comment]
That was just a report from Quickbooks I wanted you to see.
So you can see how many transactions are entered, wording and fees.
As long as your spreadsheet is sorted by Customer ID#, it will be fine.

**Comment by kbt1** [comment]
Is there any update on the Bulk Dues Entry?

**Comment by sagarjagdish** [comment]
@kbt1 : We have uploaded this change on stage, I thought I sent you update before but I didn't. 

Please verify this process on stage first.

http://71.13.159.202/aeramms_stage/duesprocessing.aspx

![image](https://user-images.githubusercontent.com/7322922/27381819-e76e54ba-56a1-11e7-9a08-9d9991cc41ce.png)

1) download template
2) fill data into template
3) upload that file and process.

This process will create new batch and you can post that batch after verification. 

Please first do that process on stage and if you have any suggestion please send us before taking action on live.

Regards,
Sagar

**Comment by omnitekinc** [comment]
@sagarjagdish Karen will be testing this on the staging server and give us feedback, she is planning to work on due processing next Monday, please move this update to Production over the weekend.

**Comment by sagarjagdish** [comment]
everything on stage and production both but it's my suggestion to take action related to bulk dues process on stage first to make sure and avoid any wrong action on production. 

Please let me know if any query.

**Comment by kbt1** [comment]
I thought all the payments would be pulled automatically into the spreadsheet so then i can edit prices and add or delete records before exporting the file into dues?
If I have to enter all the payments on the spreadsheet, that's the same as entering everything into the dues system.
I thought you would be able to pull all the monthly payments into the spreadsheet from the membership system since they are all marked and then when i download the template, all the info is there.
This really doesnt make anything easier for me.

**Comment by sagarjagdish** [comment]
We have provided option to download monthly member who having expiry in next month. Please use "Upcoming Monthly Renewal" button to download excel with member data.  Please make sure it's exported only those member who having expiry between today to 2nd of next month. 

Please review this change and let me know if any change required in this implementation.

![image](https://user-images.githubusercontent.com/7322922/27481448-425e06ec-583a-11e7-86e4-4b8405eafadf.png)

**Comment by kbt1** [comment]
Works in "Live" version but in the "STAGE" version, all expiration dates have been updated due to thats the way we used to do it.
In the "Stage Version" can you set all the monthly payment people to 6/1/17 so I can check it out.
Nothing comes up cause they are all updated.

**Comment by sagarjagdish** [comment]
We have updated logic and now it's exporting data from the stage. Please verify and let us know if any query.

**Comment by kbt1** [comment]
Works great. 
My only question is, can you have 2: **Uploading Monthly Renewal Credit Cards** and **Uploading Monthly Renewal Checks**???  It would just be easier this way.
I guess the next thing is to update it in the Live version we are using and I can create a bulk batch.

**Comment by kbt1** [comment]
Sagar
Is the above request possible?
make a credit card upload and checking upload?
let me know because I am ready to go live and start using it.

**Comment by sagarjagdish** [comment]
@kbt1 : It's possible but I have question please see below.

1) here you need 2 **download** button like "Upcoming Monthly Renewal CC" & "Upcoming Monthly Renewal Checks"?

I don't think there is 2 **upload** button required because single upload can manage both. If you upload CC member detail then it's create batch for those and if you upload excel with Checks detail then it will create batch for Check.

Please let me know your feedback on this.

**Comment by kbt1** [comment]
When I upload the batch, it pulls both credit card and checks and I do them separately each month. 
So i can save 2 spreadsheets and add or delete from them each month?
I guess I need to do the actual batch and see how it goes.
Let me know when it is ready to go on the Live database and i will work on it.
I have to manually set back all the expiration dates of the monthly accounts first. Some are paying May and June and some are paying June and July right now.

**Comment by sagarjagdish** [comment]
@kbt1  : currently all the changes which available on stage are present on live. So you can see same function on live as well.

regarding below requirement, please see my comment.
+++++++++++++++++
When I upload the batch, it pulls both credit card and checks and I do them separately each month.
+++++++++++++++++
Okay for that we can provide 2 button like as below to pull Credit Card and Checks information. Please let me know, is the work?

![image](https://user-images.githubusercontent.com/7322922/27590840-9267f4d2-5b6d-11e7-8948-3585a9cecc9b.png)

**Comment by kbt1** [comment]
that would be perfect. so the credit card will only pull credit card monthlys and the check will only pull checking.
let me know when done.

**Comment by sagarjagdish** [comment]
Okay. We will work on this and will send you update once it's done.

**Comment by kbt1** [comment]
Sagar:
the above will be good.
When you are done, can you set Monthly Credit Cards and Checks with dates of 6/1/17 and forward to 6/1/17. Then I dont have to manually change dates. I have a few that are paying for May but I can manually do those. 
So any Monthlys with expiration dates of 6/1/17 and forward, make them 6/1/17.

Thank you!

**Comment by sagarjagdish** [comment]
@kbt1 : Okay. I have noted above request and will do.

**Comment by sagarjagdish** [comment]
@kbt1 : We have done with this change and uploaded on stage and live both. Please verify and confirm our change.

+++++++++++++++++++++++++
So any Monthlys with expiration dates of 6/1/17 and forward, make them 6/1/17.
+++++++++++++++++++++++++
We have updated only those member who having expiration dates between of 6/1/17 to 6/30/2017. Please confirm our action and let me know if any change required.

**Comment by kbt1** [comment]
**All monthlys that expire 6/1/17 and ALL months forward and into 2018 need to be set back to 6/1/17.**
Before our new system, i manually changed expirations for a full year so everyone needs to be set back so I start running monthlys through the dues processing bulk system.

When dates are updated, let me know and I will work on this today.

**Comment by sagarjagdish** [comment]
@kbt1 : We have taken action as per your above comment and updated all monthly members with expire 6/1/17 who having expire greater than 6/1/17. can you please check now?

**Comment by kbt1** [comment]
i will work on it and let you know if i have any problems today.

**Comment by kbt1** [comment]
Sagar:
I pulled the report and there are not many Prosis monthlys.
**Can you make sure Membership and Prosis monthlys that expire 6/1/17 and ALL months forward and into 2018 need to be set back to 6/1/17.**
let me know when done and i will try again,

**Comment by sagarjagdish** [comment]
We have updated both Member and Prosis Pro monthly users to 6/1/17. Please export data and let us know if any issue.

**Comment by kbt1** [comment]
Worked like a charm!!!!!

I am going to do checking now. I will let you know if I have any problems.

**Comment by sagarjagdish** [comment]
That's great. Please update me if you found any issue.

**Comment by kbt1** [comment]
Sagar
All monthlys for CHECKING ONLY dates need to be changed to 6/1/17 for Prosis Pro.
DO NOT DO ANYTHING WITH RECORDS THAT ARE MARKED WITH CREDIT CARDS, THEY HAVE BEEN POSTED.

**Comment by sagarjagdish** [comment]
Okay..let me do that.

**Comment by sagarjagdish** [comment]
@kbt1 : Upload done please check and let me know if any issue.

I have updated only record which having payment method check.

**Comment by kbt1** [comment]
Sorry, can you change the check transactions for PROSIS only to 6/30/17 so they are not locked out right now.

**Comment by sagarjagdish** [comment]
It's done I set exp to 6/30/17 for the PROSIS only.

**Comment by kbt1** [comment]
thank you.

**Comment by kbt1** [comment]
Processed checking and uploaded and got this error
![image](https://user-images.githubusercontent.com/26387044/27698496-e4d046de-5cbc-11e7-8538-0576f15465e6.png)
Payment method is check
Payment term is marked as 2 or more because they are paying June and July to equal 8/1/17 exp date
and 1 member paid 10 months to pay off the monthly.
can you allow multiple terms?

**Comment by sagarjagdish** [comment]
@kbt1 : 
++++++++++
can you allow multiple terms?
++++++++++
=> Currently we are not implemented function to allow multiple terms, we will implement this function.

++++++++++
Processed checking and uploaded and got this error
++++++++++
error occur only because of blank payment method. It should not be blank.

![image](https://user-images.githubusercontent.com/7322922/27700067-16e47ca0-5d1a-11e7-95f9-be90dab63df3.png)

**Comment by kbt1** [comment]
here is my spreadsheet, 
all fields are filled in.
let me know when I can do this once you allow multiple terms.

thank you
[Monthly Checking 062917.xlsx](https://github.com/omnitekinc/AERA_MSS_Clients/files/1112892/Monthly.Checking.062917.xlsx)

once i get this batch posted for Checks, i am going to go through all records and make sure they are marked with payment types.

**Comment by kbt1** [comment]
Payment Term is invalid, i dont think it likes multiple payments for 1 company.
payment term is 1 for the monthly and i have 2, 3 and 10 for some checking.
will you be able to allow this?

**Comment by sagarjagdish** [comment]
@kb1 : 

We have resolved "Payment Term is invalid" issue for check and now it's working fine on production.

++++++++++++++
payment term is 1 for the monthly and i have 2, 3 and 10 for some checking.
will you be able to allow this?
++++++++++++++
=> currently entire system having check like if user is monthly then payment will come as monthly. If you need to allow multiple payment term then we need to lookup for the report and need to make related change. Please let us know if you want to allow multiple payment term.

This implementation will take 2 or 3 days as we need to gone through entire system.
